ONSdigital/eq-questionnaire-runner
# Fix Remove Rule Answers Bug for old rules and introduce for new

### What is the context of this PR?
This PR fixes the bug with remove answers when routing backwards. It also introduces it for the new when rules. Personally I don't think path_finder.py should be responsible for removing answers, but without a refactor I couldn't see an easy way to do it. 

### How to review 
With both test_new_confirmation_question and test_confirmation_question, make sure when you route backwards that the answer and block is removed and progress is updated to in progress. You can do this as follows

1. Launch the survey
2. For test_new_confirmation_question answer "yes" to the "Do you want the confirmation question to route backwards" question
3. Enter "0" in Total number of employees
4. Answer yes to the next question (the are slightly different for each one, but are both confirmation questions)
5. Check the JSON http://localhost:5000/dump/debug and make sure confirm-zero-employees-answer is in the answers and the block (confirm-zero-employees-block) is in the progress store. Confirm the section is COMPLETE
6. Go back, by using previous or by the browser
7. Click No or equivalent on the confirmation question
8. Now check the JSON again and make sure confirm-zero-employees-answer and confirm-zero-employees-block is gone, and the progress is IN_PROGRESS

### Checklist
